src/main/java/online/talkandtravel/service/impl/EventServiceImpl.java

```diff
@@ -105,9 +105,21 @@ public MessageDto leaveChat(EventRequest request) {
             .type(MessageType.LEAVE)
             .build();
     message = messageRepository.save(message);
+
     return messageMapper.toMessageDto(message);
   }
 
+  @Override
+  @Transactional
+  public void deleteChatIfEmpty(EventRequest request) {
+    Chat chat = getChat(request);
+    if (chat.getChatType().equals(ChatType.PRIVATE)) {
```

Comment:
you could use && to prevent using nested if statement

like: 
if ( chat.getChatType().equals(ChatType.PRIVATE) &&
        chat.getUsers().isEmpty() ) {
    // code...
}